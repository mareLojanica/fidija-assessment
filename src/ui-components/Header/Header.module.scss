/* HEADER */
.homepage__header {
  background-color: #e0e0e0;
  display: flex;
  justify-content: start;
  padding: 30px 100px;
  position: relative;
  z-index: 2;

  @media (max-width: 767px) {
    padding: 40px 15px;
  }
}
/* HEADER CONTENT */
.homepage__header-content {
  max-width: 550px;
}
/* TITLE */
.homepage__title {
  font-size: 32px;
  font-weight: bold;
  margin: 0;
  color: #222;
  text-transform: uppercase;
  letter-spacing: 1px;
  margin-bottom: 1.5rem;
  @media (max-width: 767px) {
    font-size: 24px;
  }
}
/* SUBTITLE */
.homepage__subtitle {
  font-size: 18px;
  color: #828181;
  margin-top: 15px;
  line-height: 1.6;
  font-weight: 400;
  max-width: 440px;
  @media (max-width: 767px) {
    font-size: 16px;
  }
}
