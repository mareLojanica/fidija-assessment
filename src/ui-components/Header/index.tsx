import React, { FC, ReactNode } from "react"
import styles from "./Header.module.scss"

const Header: FC<{
	title: string
	description?: ReactNode
	showInfo?: ReactNode
}> = ({ title, description, showInfo }) => {
	return (
		<header className={styles["homepage__header"]}>
			<div className={styles["homepage__header-content"]}>
				<h1 className={styles["homepage__title"]}>{title}</h1>
				<div className={styles["homepage__subtitle"]}>
					{description}
				</div>
			</div>
			{showInfo}
		</header>
	)
}

export default Header
