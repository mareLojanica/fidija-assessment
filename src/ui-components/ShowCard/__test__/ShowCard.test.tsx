import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../../Header";
import "@testing-library/jest-dom";

describe("Header Component", () => {
  test("renders title", () => {
    render(<Header title="TV Blend" />);
    expect(screen.getByText("TV Blend")).toBeInTheDocument();
  });

  test("renders description when provided", () => {
    render(
      <Header
        title="TV Blend"
        description={
          <>
            <p>TV Show and web series database.</p>
            <p>
              Create personalized schedules. Episode guide, cast, crew, and
              character information.
            </p>
          </>
        }
      />
    );

    expect(
      screen.getByText("TV Show and web series database.")
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        "Create personalized schedules. Episode guide, cast, crew, and character information."
      )
    ).toBeInTheDocument();
  });

  test("renders showInfo when provided", () => {
    const mockShowInfo = <div data-testid="show-info">Show Details Here</div>;

    render(<Header title="TV Blend" showInfo={mockShowInfo} />);

    expect(screen.getByTestId("show-info")).toBeInTheDocument();
    expect(screen.getByTestId("show-info")).toHaveTextContent(
      "Show Details Here"
    );
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <Header
        title="TV Blend"
        description={<p>TV Show and web series database.</p>}
        showInfo={<div data-testid="show-info">Show Details Here</div>}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
