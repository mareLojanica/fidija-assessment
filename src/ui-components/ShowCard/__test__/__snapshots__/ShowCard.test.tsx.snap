// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="homepage__header"
  >
    <div
      class="homepage__header-content"
    >
      <h1
        class="homepage__title"
      >
        TV Blend
      </h1>
      <div
        class="homepage__subtitle"
      >
        <p>
          TV Show and web series database.
        </p>
      </div>
    </div>
    <div
      data-testid="show-info"
    >
      Show Details Here
    </div>
  </header>
</DocumentFragment>
`;
